cmd/cloudFoundryCreateService.go

```diff
@@ -0,0 +1,115 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/SAP/jenkins-library/pkg/cloudfoundry"
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/pkg/errors"
+)
+
+func cloudFoundryCreateService(config cloudFoundryCreateServiceOptions, telemetryData *telemetry.CustomData) {
+
+	cf := cloudfoundry.CFUtils{Exec: &command.Command{}}
+
+	err := runCloudFoundryCreateService(&config, telemetryData, cf)
+
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+
+}
+
+func runCloudFoundryCreateService(config *cloudFoundryCreateServiceOptions, telemetryData *telemetry.CustomData, cf cloudfoundry.CFUtils) error {
+
+	var err error
+
+	var c = cf.Exec
+
+	loginOptions := cloudfoundry.LoginOptions{
+		CfAPIEndpoint: config.CfAPIEndpoint,
+		CfOrg:         config.CfOrg,
+		CfSpace:       config.CfSpace,
+		Username:      config.Username,
+		Password:      config.Password,
+	}
+
+	err = cf.Login(loginOptions)
+
+	if err != nil {
+		return fmt.Errorf("Error while logging in: %w", err)
+	}
+
+	if err == nil {
+		err = cloudFoundryCreateServiceRequest(config, telemetryData, c)
+	}
+
+	if err != nil {
+		return fmt.Errorf("Service creation failed: %w", err)
+	}
+
+	defer func() {
```

Comment:
The logout does not happen in case there was a successful login, but `cloudFoundryCreateServiceRequest` failed. But also in this case we should perform the logout. In order to get this running the defer function needs to be some lines above, right after the `err` check for the login.

Reply:
Placed it right after `err` check now...